<html>
  <head>
    <title>Single cell biology</title>
    <link rel="stylesheet" href="css/reveal.css">
    <link rel="stylesheet" href="css/theme/white.css">
    <link rel="stylesheet" href="lib/css/zenburn.css">

    <script>
      var link = document.createElement( 'link' );
      link.rel = 'stylesheet';
      link.type = 'text/css';
      link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
      document.getElementsByTagName( 'head' )[0].appendChild( link );
    </script>
  </head>

  <body>
    <div class="reveal">
      <div class="slides">
        <section>
          <h3> Single cell biology: <br >Studying every cell</h3>
          <iframe width="900" height="700" frameborder="0" scrolling="no" src="https://plot.ly/~mbolisetty/1567.embed"></iframe>
        </section>

        <section>
          <h3> What is a cell? </h3>
            <p>
              <blockquote >
                &ldquo;The cell is the most basic unit of life&rdquo;
                <br>
                  -Theodor Schwann
              </blockquote>

              <img src="img/Wilson1900Fig2.jpg" width="50%" style="background:none; border:none; box-shadow:none; ">
            </p>
        </section>

        <section>
          <section>
            <h2> Early recognition of different functional cell types </h2>
          </section>

          <section>
            <p>
              Retina (neuronal theory) by Ramon y Cajal 1852 - 1934
            </p>
            <img src="img/Cajal2.jpg" width="50%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <p>
              Many cell types were simply identified by looking under a
              microscope and looking at morphology. Technological advances in
              microscopes made this easier and faster.
            </p>
          </section>

          <section>
            <p>
              Up until recently, it was difficult or impossible to classify
              even modest numbers of cells by anything other than cell surface
              markers or observed functions.
            </p>
            <p>
              Advances in sequencing afford the ability to now look at
              transcriptional activity of many cells at a time.
            </p>
          </section>
        </section>

        <section>
          <section>
            <h4>Experimental methods to generate scRNA-seq data:</h4>
            <ul>
              <li>CEL-seq (1,2)</li>
              <li>Drop-seq</li>
              <li>InDrop</li>
              <li>MARS</li>
              <li>SCRB</li>
              <li>Smart-seq (1,2)</li>
              <li>SMARter</li>
              <li>10X-Chromium</li>
            </ul>
          </section>

          <section>
            <h3> Accuracy of different approaches </h3>
            <br>
            <img src="img/accuracy.png" width="100%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <h3> Sensitivity of different approaches </h3>
            <br>
            <img src="img/sensitivity.png" width="100%" style="background:none; border:none; box-shadow:none; ">
          </section>
        </section>

        <section>
          <section>
            <h3> 10X Genomics </h3>
            <br>
            <img src="img/10xgenomics.png" width="100%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <h3> Microfluidics enables us to capture each cell and look at all the genes</h3>
            <iframe frameborder=0 width=95% height=586 marginheight=0 marginwidth=0 scrolling=no data-src="img/mmc4.mp4"></iframe>
          </section>

          <section>
            <h3>Our goal today:</h3>
            <img src="img/smoothie.jpg" width="40%" style="background:none; border:none; box-shadow:none; ">

            <p>Until 6 years ago, we only had the ability to profile a tissue as bulk.</p>
          </section>

          <section>
            <img src="img/fruit.jpg" width="50%" style="background:none; border:none; box-shadow:none; ">
            <p>
              But now we can examine each piece individually.
            </p>
            <p style="font-size: 12pt">
              Analogy from Hyun Min Kang, UMichigan, 2018
            </p>
          </section>
        </section>

        <section>
          <section>
            <h3> Bioinformatics </h3>
            <br>
            <br>
            <h4> Reads -> Matrix -> Insights</h4>
          </section>

          <section>
            <h3> Reads -> Matrix </h3>
            <br>
            <h4> CellRanger Pipeline</h4>
            <img src="img/pipeline.png" width="150%" style="background:none; border:none; box-shadow:none; ">
          </section>


          <section>
            <h3> Read 1: Cell barcode UMI </h3>
            <br>
            <img src="img/read1-bill.png" width="100%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <h3> Read 2: Transcript specific read </h3>
            <br>
            <img src="img/read2-bill.png" width="100%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <h3> BAM: Alignment and transcript assignment </h3>
            <br>
            <img src="img/bam-bill.png" width="200%" style="background:none; border:none; box-shadow:none; ">
          </section>
        </section>

        <section>
          <section>
            <h3> Summary stats </h3>
            <img src="img/summaryStats.png" width="50%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <h3> Summary stats </h3>
            <img src="img/summaryStats2.png" width="50%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <h3> Expression matrix </h3>
            <br>
            <img src="img/expression.png" width="50%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <p> Data is sparse - lots and lots of zeros </p>
            <img src="img/ZerosIn10XData.png" width="50%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <p>Single cell RNA-seq data is described as a Zero-Inflated negative binomal distribution</p>
            <img src="img/zi-nb.png" width="60%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <p>In comparison to a bulk RNA-seq distribution</p>
            <img src="img/nb.png" width="60%" style="background:none; border:none; box-shadow:none; ">
          </section>

          <section>
            <h3> Dropout rates </h3>
            <p>
              These models introduce a new parameter  dd , for the dropout rate,
              to the negative binomial model. The dropout
              rate of a gene is strongly correlated with the mean expression of
              the gene.
            </p>
          </section>

          <section>
            <h3> Detection Probability </h3>
            <img src="img/DetectionAsAFunctionOfExpression.png" width="60%" style="background:none; border:none; box-shadow:none; ">
          </section>
        </section>

        <section>
          <section>
            <h3>Matrix -> Insights</h3>
            <p>
              The primary goal of every single cell experiment is to describe
              the heterogeneity of cell types present in the starting tissue
              population.
            </p>
          </section>

          <section>
            <h3>Challenges</h3>
            <p>
              Exponential growth of new methodologies to generate, process, and
              analysis scRNA-seq data.
            </p>
            <p>
              Many methods rely on heuristics or empirical results, and lack
              proper validation with existing methods.
            </p>
          </section>

          <section>
            <h3>9/10 Analysts agree</h3>
            <p>
              The majority of analysis methods roughly agree with the following
              pipeline:
            </p>
            <ul>
              <li>Filtering low quality cells</li>
              <li>Filtering really sparsely expressed genes</li>
              <li>Normalize gene expression</li>
              <li>Feature selection for clustering</li>
              <li>Dimensionality reduction</li>
              <li>Cluster identification</li>
              <li>Marker gene identification</li>
              <li>Cell type inference</li>
            </ul>
          </section>

          <section>
            <h3>Analysis Challenges</h3>
            How to...
            <ul>
              <li>discriminate lowly expressing cells from ambient mRNA?</li>
              <li>reliably call expressed genes?</li>
              <li>cluster cells without bias?</li>
              <li>id cell types without clustering?</li>
              <li>infer celltypes from marker genes? without biological insight?</li>
            </ul>
          </section>

          <section>
            <h3>Philosphical Challenges</h3>
            <ul>
              <li>How to define a "cell type"?</li>
              <li>Do cells that cluster together always belong to the same "type"?</li>
            </ul>
          </section>

          <section>
            <h3>Let's dive in!</h3>
            <p>
              Let's open the e-notebooks and make sure you have everything you
              need
            </p>
            <p>
              Each of you has an Amazon EC2 cloud instance with Jupyter notebooks
              running Python3 kernels
            </p>
          </section>
        </section>

        <section>
          <section>
            <h3>Analysis details (optional)</h3>
          </section>

          <section>
            <h3>Differential Expression (DE) analysis</h3>
            <p>
              Every analyses with analyzing bulk RNA-seq data eventually
              involves to identifying differentially expressed genes between
              experimental conditions.  Many methods, e.g. DESeq2 and edgeR,
              have been developed for bulk RNA-seq. These algorithms although
              not ideal for single cell RNA-seq data can still provide robust
              results.
            </p>
          </section>

          <section>
            <p>Some of the many algorithms are listed here for reference:
            <ul>
              <li>edgeR</li>
              <li>DESeq2</li>
              <li>KS</li>
              <li>Mann whitney U</li>
              <li>T-test</li>
              <li>SCDE - specifically for single cell</li>
              <li>T-test</li>
              <li>AUROC</li>
              <li>And many more</li>
            </ul></p>
          </section>

          <section>
            <h3> Marker genes </h3>
            <p>
              Plp1, Cldn11, Ugt8a, Sirt2, Mag, Mobp, Cnp, Mbp, Cck, Pcp4,
              Nap1l5, Meg3, Gap43, Rgs5, Itm2a, Tmsb4x, Slc16a1, Igfbp7, Hbb-bt,
              Hbb-bs, Bsg, Klf2, Junb, Hba-a1, Hba-a2, Sparc, Ctla2a, Cldn5,
              Dbi, Atp1a2, Slc1a2, Cst3, Fabp5, Plpp3, Id3, Sparcl1, Ptn, Apoe,
              Gja1, Fabp7, Cpe, Mt3, Mt1, Ndrg2, Clu, Aldoc, Id2, Ckb, Pla2g7,
              Aqp4, Col3a1, Igfbp2, Ptgds, S100a6, Col1a2, Ptn, Cxcl12, Nupr1,
              Igf2, Dcn, Lum, Col4a1, Klf2, Junb, Sparc, Vtn, Col1a1, Lgals1,
              Apod, Fcer1g, B2m, Cst3, Tmsb4x, P2ry12, Ctss, C1qb, C1qc , C1qa,
              Tyrobp, Ftl1, Ctsd, Junb, Ccl2, Ccl7, Ccl12, Ccl3 , Ccl4, Cd83,
              Hexb, Rps29, Lgmn, Sepp1
            </p>
          </section>

          <section>
            <h3>Cell type inference</h3>
            <ul>
              <li>Fabp7 (BLBP) -  astrocytes</li>
              <li>Pcp4 - Purkinje neurons</li>
              <li></li>
            </ul>
          </section>
        </section>

        <section>
          <h3>CellView</h3>
          <p>
            A <a href="https://doi.org/10.1101/123810">web application</a> to
            visualize <br> and explore single cell RNA-seq data.
          </p>

          <pre><code data-trim contenteditable>
spl.write_rds(cpt, feature_data, cluster_data)
          </code></pre>

          <p> Download Data.Rds and we will upload to this website:
          <br>
          <a href="https://mbolisetty.shinyapps.io/CellView/">
            https://mbolisetty.shinyapps.io/CellView</a>
          <br>
          <br>
          Or if within the JAX network:
          <br>
          <a href="http://ctshiny01.jax.org/flynnb/CellView/">
          http://ctshiny01.jax.org/flynnb/CellView/</a>
          </p>
        </section>

        <section>
          <section>
            <h3> 2,300,000 and counting </h3>
            <iframe width="600" height="400" frameborder="0" scrolling="no" data-src="https://plot.ly/~mbolisetty/1562.embed"></iframe>
          </section>
        </section>

        <section>
          <h3> Other resources</h3>
          <ul>
            <li><a href="http://satijalab.org/seurat/">Seurat analysis package
                (R)</a></li>
            <li><a
                href="https://hemberg-lab.github.io/scRNA.seq.course/index.html">
                Hemberg Lab scRNA-seq Analysis e-Book (Sanger Institute)</a></li>
            <li><a
                href="https://www.10xgenomics.com/10x-university/single-cell/">
                10X Genomics University resources</a></li>
            <li><a
                href="https://github.com/seandavi/awesome-single-cell">
                Awesome Single-Cell repository (Sean Davis)</a></li>
          </ul>
        </section>

        <section>
          <h3>Acknowledgments </h3>
          <br>
          <ul>
            <li>Sandeep Namburi - Research IT</li>
            <li>Aditya Kovuri - Research IT </li>
            <li>Mohan Bolisetty - Single cell </li>
            <li>Paul Robson - Single cell </li>
            <li>Mike Samuels - Single cell </li>
            <li>Genome Technologies</li>
          </ul>
        </section>

        <section>
          <h3> Questions? </h3>
            <br>
            <br>
            <p>
              email me:
              <br>
              <a href="mailto:bill.flynn@jax.org">bill.flynn@jax.org</a>
            </p>
            <p>
              slides:
              <br>
              <a href="wflynny.github.io/JAX-NeuroGeneticsSClDataAnalysis/">
                wflynny.github.io/JAX-NeuroGeneticsSClDataAnalysis/</a>
            </p>
        </section>
      </div>
    </div>
    <script src="js/reveal.js"></script>
    <script>
      Reveal.initialize();
    </script>
  </body>
</html>
